app/core/models.py

```diff
@@ -0,0 +1,89 @@
+import uuid
+import os
+from django.db import models
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager, \
+    PermissionsMixin
+
+from django.conf import settings
+
+
+def recipe_image_file_path(instance, filename):
+    """Generate filepath for recipe image"""
+    ext = filename.split('.')[-1]
+    filename = f'{uuid.uuid4()}.{ext}'
+    return os.path.join('uploads/recipe/', filename)
+
+
+class UserManager(BaseUserManager):
+    def create_user(self, email, password=None, **extra_fields):
+        """Creates and saves a new user"""
+        if not email:
+            raise ValueError('Users must have an email address')
```

Comment:
User goes beyond the scope of this exercise, so as a comment:

Why don't you enforce email to not be empty in the definition of the column?

Reply:
The blank flag in the user email model field will not allow empty values entering through the API.

On the other hand, the null flag will set the NOT NULL constraint on the psql email column. There is no NOT BLANK postgres constraint so you could insert an empty string as email.

The check is there to protect the private api, not allowing developers to build invalid (but database valid) user objects.

What do you think? I'm new to python so it's highly probable im missing on something.